### PR TITLE
Fix: Support case insensitive tags (fixes #163)

### LINKED ISSUE #163: Support case insensitive tags
**What version are you using?**
master

**What JSDoc comment were you trying to parse?**
/** @Const {String} constname */

**What Doctrine options were you using (sloppy, unwrap, etc.)?**
{ unwrap: true }

**What happened? Include output if possible.**
Tag gets parsed as 
```
{ description: '{String} constname', title: 'Const' }
```

**What did you expect to happen?**
Tag gets parsed as 
```
{
  description: null,
  name: 'constname',
  title: 'Const',
  type: { name: 'String', type: 'NameExpression' }
}
```

See eslint related issues: https://github.com/eslint/eslint/issues/6121 https://github.com/eslint/eslint/issues/6145

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: lib/doctrine.js [hunk 1]
```diff
@@ -404,7 +404,7 @@
 
         function TagParser(options, title) {
             this._options = options;
-            this._title = title;
+            this._title = title.toLowerCase();
```
Comment (nzakas) [inline lib/doctrine.js]: Hmmm, I'm not sure this is the right thing to do. Swallowing letter casing isn't really what parsers do.

Comment (nzakas) [inline lib/doctrine.js]: Doh, nevermind. I read this wrong.

**Comment by nzakas** [comment]
Lgtm

**Comment by alberto** [comment]
Thanks. Do you intend to do a release so we can update it in eslint? It's causing parsing errors

**Comment by nzakas** [comment]
Yes, I'll do a release this week.

**Pull request merged**